db/migrate/20240715073605_add_smtp_ssl_mode_to_groups.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+class AddSmtpSslModeToGroups < ActiveRecord::Migration[7.1]
+  def up
+    add_column :groups, :smtp_ssl_mode, :integer, default: 0, null: false
+
+    execute <<~SQL
+      UPDATE groups SET smtp_ssl_mode = (CASE WHEN smtp_ssl THEN 2 ELSE 0 END)
```

Comment:
So here we are setting the mode to STARTTLS if the old column was set to true, because that is the behaviour we did here:

https://github.com/discourse/discourse/pull/27932/files#diff-4824074106447332571ce4a5b5314493f8648bbe237d5dacae965fe3886a1ad6L21

We didn't ever actually set `enable_ssl` to true in the group SMTP mailer